### PR TITLE
Feature/my map

### PR DESCRIPTION
**Pull request by RomiKMoveo**
### Description
This pull request introduces my map component to my pokemon mini site

### Changes Made
- Generate a new API key for your website
- Build a new page on your website called “My Map”.
- Show a map in the center of the page based on google maps
- Add Marker with the location of the office
- Add Autocomplete field for searching an address, when choosing an address add a new marker to the map 
- Add style map to my map 
- Add a “direction button”, when clicked shows the directions for a car from my home to the office

 ### Related Issues
- Marker was deprecated and I could not change my code to AdvancedMarkerElement 

Enter related issues here

I have:
- [ ] deleted all irrelevant comments, logs and debuggers.
- [ ] assure that I don't print full error stacks, full body response or headers from the server in the console.
- [ ] deleted all sensitive information from the code because it's being imported from environment variables.
- [ ] encrypted keys/passwords that are sent through the internet.
- [ ] sanitized all user inputs to prevent attacks.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: pokemon-minisite/package.json [hunk 1]
```diff
@@ -16,10 +16,12 @@
     "@angular/compiler": "^17.3.0",
     "@angular/core": "^17.3.0",
     "@angular/forms": "^17.3.0",
+    "@angular/google-maps": "^17.3.6",
```
Comment (noyflaysher) [inline pokemon-minisite/package.json]: do you use it ?

File: pokemon-minisite/package.json [hunk 2]
```diff
@@ -16,10 +16,12 @@
...
     "@angular/material": "^17.3.5",
     "@angular/platform-browser": "^17.3.0",
     "@angular/platform-browser-dynamic": "^17.3.0",
     "@angular/router": "^17.3.0",
+    "@googlemaps/js-api-loader": "^1.16.6",
```
Comment (noyflaysher) [inline pokemon-minisite/package.json]: the same

File: pokemon-minisite/src/app/app.module.ts [hunk 3]
```diff
@@ -18,15 +23,18 @@ import { FilterPokemonByTypeAndNamePipe } from './pipes/filter-pokemon.pipe';
...
     ],
     imports:[
         BrowserModule, 
         CommonModule,
         MatDialogModule,
         FormsModule,
         RouterModule,
-        RouterOutlet
+        RouterOutlet,
+        GoogleMapsModule,
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/app.module.ts]: you don't need to use it

File: pokemon-minisite/src/app/app.routes.ts [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+import { MyMapComponent } from './components/my-map/my-map.component';
+
+
 
 export const routes: Routes = [
     { path: '', redirectTo:'login', pathMatch:'full'},
     { path: 'pokemons', component: PokemonListComponent },
     { path: 'history', component: HistoryComponent },
     { path: 'login', component: LoginComponent},
+    { path: 'myMap', component: MyMapComponent},
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/app.routes.ts]: change to map

File: pokemon-minisite/src/app/components/header/header.component.scss [hunk 5]
```diff
@@ -4,7 +4,7 @@
 .header {
   color: $ground;
   background-color: $fire-light;
-  background-image: url("../../../assets/images/62cf234679dbabe18fa50a1e_pokeapi_256\ 1.svg");
+  background-image: url("../../../assets/images/pokemon_logo.png");
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/header/header.component.scss:7]: try to use ./assets....

File: pokemon-minisite/src/app/components/my-map/my-map-style.ts [hunk 6]
```diff
@@ -0,0 +1,284 @@
+export const myMapStyle: any[] =
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map-style.ts]: Move this file into "constants" folder in my-map folder

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+@use '../../style/variables' as *;
+@use '../../style/mixin' as *;
+
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.scss:3]: use nesting

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+span{
+    display: flex;
+    justify-content: space-between;
+    margin: 20px;
+}
+
+.buttons-container{
+    display: flex;
+    flex-direction: row;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.scss]: You don't need this. its the default direction

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    height: 40px;
+    padding: 10px;
+    font-size: 15px;
+    border: none;
+    border-radius: 4px;
+    margin-right: 10px;
+    line-height: 20px; 
+    cursor: pointer;
+    &:hover{
+        background-color: rgb(216, 228, 216);
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.scss]: use variable

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    }
+    
+}
+
+div {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.scss]: use the class name

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+  MatInputModule,
+  MatSlideToggleModule,
+  FormsModule
+],
+templateUrl: './my-map.component.html',
+styleUrl: './my-map.component.scss'
+})
+export class MyMapComponent implements AfterViewInit{
+  @ViewChild('mapContainer', { static: false }) 
+  gmap!: ElementRef;
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: Use a more meaningful name

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+
+  map!: google.maps.Map;
+  lat = 32.0624076;
+  lng = 34.7707653;
+
+  coordinates = new google.maps.LatLng(this.lat, this.lng);
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: do it in the initialize function

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+
+  mapOptions: google.maps.MapOptions = {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: the same

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+    center: this.coordinates,
+    zoom: 15
+  };
+
+  marker = new google.maps.Marker({
+    position: this.coordinates,
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: the same

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+    map: this.map,
+    title: "Office"
+  });
+  
+  @Input()  placeholder=""; 
+  @ViewChild('searchInput')
+  searchInput!: ElementRef;
+  autocomplete: google.maps.places.Autocomplete | undefined;
+
+  private directionsService = new google.maps.DirectionsService();
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: the same

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+  private directionsRenderer = new google.maps.DirectionsRenderer();
+  readonly homeCoordinates = {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/components/my-map/my-map.component.ts]: Use it in the function  you need

File: pokemon-minisite/src/app/interfaces/addres-search-result.interface.ts [hunk 17]
```diff
@@ -0,0 +1,7 @@
+export interface AddresSearchResult {
```
Comment (noyflaysher) [inline pokemon-minisite/src/app/interfaces/addres-search-result.interface.ts]: Do you use it ?

File: pokemon-minisite/package.json (hunk 1 above)
Comment (RomiKMoveo) [reply pokemon-minisite/package.json]: done

File: pokemon-minisite/package.json (hunk 2 above)
Comment (RomiKMoveo) [reply pokemon-minisite/package.json]: done

File: pokemon-minisite/src/app/app.module.ts (hunk 3 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/app.module.ts]: done

File: pokemon-minisite/src/app/app.routes.ts (hunk 4 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/app.routes.ts]: done

File: pokemon-minisite/src/app/components/header/header.component.scss (hunk 5 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.scss:7]: not working any other way

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss (hunk 7 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.scss:3]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss (hunk 8 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.scss]: I need it ):

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss (hunk 9 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.scss]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.scss (hunk 10 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.scss]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 11 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 12 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 13 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 14 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 16 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

File: pokemon-minisite/src/app/components/my-map/my-map.component.ts (hunk 15 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/my-map/my-map.component.ts]: done

**Pull request merged**